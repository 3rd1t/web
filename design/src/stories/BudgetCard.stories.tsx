import React from "react";
import { BudgetCard } from '../components/BudgetCard';

export default {
  title: "Budget Card",
};

export const card = () => {

  return (
    <BudgetCard  expensesCount= {195} totalBudget={1500.00} start={new Date()} end={new Date()} budgeted={4.99} expenses={625.12} income={10.00}/>
  )
};
