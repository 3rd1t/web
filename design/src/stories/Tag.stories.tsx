import React from 'react';
import Tag from "../components/Tag"
export default { title: 'Tag' };

export const Default = () => {
    return (
        <Tag>Hello Tag</Tag>
    )
} 
