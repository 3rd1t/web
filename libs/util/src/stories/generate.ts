/**
 * Return all components from storybook.
 *
 * @example
 * import * as stories from "./<component>.stories"
 *
 * const testCases = generateTestCasesFromStories(stories)
 * testCases.forEach((s) => {
 *   it(s.name + "should match snapshot", () => {
 *     const { baseElement } = render(s.component())
 *     expect(baseElement).toMatchSnapshot()
 *   })
 * })
 *
 *
 * @param stories - import * as stories from "./<component>.stories"
 * @returns Name and render function for each component
 */
export function generateTestCasesFromStories(
  stories: Record<string, any>, // eslint-disable-line
): { name: string; component: () => HTMLElement }[] {
  return Object.entries(stories)
    .filter((s) => s[0] !== "default")
    .map((s) => {
      return {
        name: s[0],
        component: s[1],
      }
    })
}
