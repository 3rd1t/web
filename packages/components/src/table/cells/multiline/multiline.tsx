import React from "react"
import { Wrapper } from "../wrapper/wrapper"

/* eslint-disable-next-line */
export interface MultilineProps {
  title: string | number
  content: string | number
  key?: number
}

export const Multiline = ({ title, content, key }: MultilineProps) => {
  return (
    <Wrapper>
      <span className="text-sm leading-5 text-gray-900">{title}</span>
      <p className="text-sm leading-5 text-gray-500">{content}</p>
    </Wrapper>
  )
}

export default Multiline
