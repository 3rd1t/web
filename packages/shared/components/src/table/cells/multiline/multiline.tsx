import React from "react"

/* eslint-disable-next-line */
export interface MultilineProps {
  title: string | number
  content: string | number
  key?: number
}

export const Multiline = ({ title, content, key }: MultilineProps) => {
  return (
    <td key={key} className="px-6 py-4">
      <span className="text-sm leading-5 text-gray-900">{title}</span>
      <p className="text-sm leading-5 text-gray-500">{content}</p>
    </td>
  )
}

export default Multiline
