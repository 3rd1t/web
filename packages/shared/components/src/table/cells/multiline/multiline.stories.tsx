import React from "react"
import { Multiline, MultilineProps } from "./multiline"

export default {
  component: Multiline,
  title: "Multiline",
}

export const primary = () => {
  /* eslint-disable-next-line */
  const props: MultilineProps = {}

  return <Multiline />
}
