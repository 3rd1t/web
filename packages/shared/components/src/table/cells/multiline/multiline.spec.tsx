import { render } from "@testing-library/react"
import * as stories from "./multiline.stories"

describe("Multiline", () => {
  describe("Stories", () => {
    Object.entries(stories).forEach((s) => {
      if (s[0] !== "default") {
        it(s[0] + "should match snapshot", () => {
          const { baseElement } = render(s[1]())
          expect(baseElement).toMatchSnapshot()
        })
      }
    })
  })
})
