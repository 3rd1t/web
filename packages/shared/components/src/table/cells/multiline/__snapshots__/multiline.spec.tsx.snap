// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiline Stories primaryshould match snapshot 1`] = `
<body>
  <div>
    <td
      class="px-6 py-4"
    >
      <span
        class="text-sm leading-5 text-gray-900"
      >
        Title
      </span>
      <p
        class="text-sm leading-5 text-gray-500"
      >
        Content
      </p>
    </td>
  </div>
</body>
`;
